lib/components/admin/trip-status.js

```diff
@@ -6,53 +6,80 @@ import { connect } from 'react-redux'
 import * as fieldTripActions from '../../actions/field-trip'
 import * as formActions from '../../actions/form'
 import Icon from '../narrative/icon'
+import { getTripFromRequest } from '../../util/call-taker'
+
 import {
   Bold,
   Button,
   Full,
   Header,
   Para
 } from './styled'
-import { getTripFromRequest } from '../../util/call-taker'
 
 class TripStatus extends Component {
-  _getTrip = () => getTripFromRequest(this.props.request, this.props.outbound)
-
   _formatTime = (time) => moment(time).format(this.props.timeFormat)
 
-  _formatTripStatus = () => {
-    if (!this._getStatus()) {
-      return (
-        <span>
-          No itineraries planned! Click Plan to plan trip.
-        </span>
-      )
+  _onDeleteTrip = () => {
+    const { deleteRequestTripItineraries, request, trip } = this.props
+    if (!confirm('Are you sure you want to delete the planned trip?')) {
+      return
     }
-    const trip = this._getTrip()
-    if (!trip) return <span>Error finding trip!</span>
-    return (
-      <span>
-        {trip.groupItineraries.length} group itineraries, planned by{' '}
-        {trip.createdBy} at {trip.timeStamp}
-      </span>
-    )
+    deleteRequestTripItineraries(request, trip.id)
+  }
+
+  _onPlanTrip = () => {
+    const { outbound, planTrip, request, status, trip } = this.props
+    if (status && trip) {
+      if (!confirm('Re-planning this trip will cause the trip planner to avoid the currently saved trip. Are you sure you want to continue?')) {
+        return
+      }
+    }
+    planTrip(request, outbound)
+  }
+
+  _onSaveTrip = () => {
+    const { outbound, request, saveRequestTripItineraries } = this.props
+    saveRequestTripItineraries(request, outbound)
   }
 
-  _getStatus = () => {
-    const {outbound, request} = this.props
-    return outbound ? request.outboundTripStatus : request.inboundTripStatus
+  _onViewTrip = () => {
+    const { outbound, request, viewRequestTripItineraries } = this.props
+    viewRequestTripItineraries(request, outbound)
   }
 
-  _getStatusIcon = () => this._getStatus()
+  _renderStatusIcon = () => this.props.status
     ? <Icon type='check' className='text-success' />
     : <Icon type='exclamation-circle' className='text-warning' />
 
-  _onPlanTrip = () => this.props.planTrip(this.props.request, this.props.outbound)
-
-  _onSaveTrip = () => this.props.saveRequestTrip(this.props.request, this.props.outbound)
+  _renderTripStatus = () => {
+    const { status, trip } = this.props
+    if (!status || !trip) {
+      return (
+        <Para>
+          <span>
```

Comment:
Remove the extra `<span>`.

Reply:
Removed in e381aa802e86af0e6889e57e83cec419af1d5589